pkg/providers/version/suite_test.go

```diff
@@ -70,9 +87,61 @@ var _ = AfterEach(func() {
 })
 
 var _ = Describe("Operator", func() {
-	It("should resolve Kubernetes Version via Describe Cluster", func() {
-		endpoint, err := awsEnv.VersionProvider.Get(ctx)
-		Expect(err).ToNot(HaveOccurred())
-		Expect(endpoint).To(Equal("1.30"))
+
+	Context("with EKS_CONTROL_PLANE=true", func() {
+		BeforeEach(func() {
+			versionProvider = version.NewDefaultProvider(env.KubernetesInterface, cache.New(awscache.DefaultTTL, awscache.DefaultCleanupInterval), fakeEKSAPI)
+			os.Setenv("EKS_CONTROL_PLANE", "true")
+		})
+
+		It("should resolve Kubernetes Version via Describe Cluster with no errors", func() {
+			endpoint, err := versionProvider.Get(ctx)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(endpoint).To(Equal("1.29"))
+		})
+
+		It("should handle EKS API errors and fallback to K8s API", func() {
```

Comment:
Discussed offline, but we shouldn't fallback to the k8s API if we have insufficient permissions. We should require that `DescribeCluster` permissions be granted if a user specifies that they're running on an EKS control plane.